### PR TITLE
[TT-10264] Added readme for enterprise portal

### PR DESCRIPTION
**Pull request by komalsukhani**
## Description

Added readme for enterprise portal 

## Related Issue

[TT-10264](https://tyktech.atlassian.net/browse/TT-10264)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)
- [x] Documentation updates or improvements.

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [x] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
     - [ ] I have manually updated the README(s)/documentation accordingly.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

[TT-10264]: https://tyktech.atlassian.net/browse/TT-10264?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: components/tyk-dev-portal/README.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+The fields `TYK_ORG` and `TYK_AUTH` are the Tyk Dashboard _Organisation ID_ and the Tyk Dashboard API _Access Credentials_ respectively. These can be obtained under your profile in the Tyk Dashboard.
+
+### 2. Install chart
+To install the chart from the Helm repository in namespace `tyk` with the release name `tyk-dev-portal`:
+```bash
+helm repo add tyk-helm https://helm.tyk.io/public/helm/charts/
+helm show values tyk-helm/tyk-dev-portal > values.yaml 
+```
+Note: 
+* Set license key at `.Values.license`
```
Comment (buraksekili) [inline components/tyk-dev-portal/README.md:41]: Maybe, we can use GitHub markdown utilities for `Note:`
```suggestion
> [!NOTE]
> Set license key at `.Values.license`
```

File: components/tyk-dev-portal/README.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+ By default, Helm chart will try to discover Dashboard service in same namespace. If you want to explicity specify dashboard url, you can set `.Values.overrideTykDashUrl` in values.yaml file.
+
+```bash
+helm install tyk-dev-portal tyk-helm/tyk-dev-portal -n tyk --create-namespace -f values.yaml
+```
+## Uninstalling the Chart
+```bash
+helm uninstall tyk-dev-portal -n tyk
+```
+
```
Comment (buraksekili) [inline components/tyk-dev-portal/README.md:52]:
```suggestion
```

File: components/tyk-dev-portal/README.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    
+You can update any value in your local values.yaml file and use `-f [filename]` flag to override default values during installation. Alternatively, you can use `--set` flag to set it during chart installation.
+
+### Tyk Developer Enterprise Portal License (Required)
+
+Tyk Developer Enterprise Portal License is required. It can be set up in `license` or through secret `useSecretName`. The secret should contain a key called `DevPortalLicense`.
+
+```yaml
+  # Developer Portal license.
+  license: ""
```
Comment (buraksekili) [inline components/tyk-dev-portal/README.md:75]:
```suggestion
# Developer Portal license.
license: ""
```

File: components/tyk-dev-portal/README.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+      - ReadWriteOnce
+    size: 8Gi
+    annotations: {}
+    labels: {}
+    selector: {}
+```
+
+### Database Settings
+Portal uses database to store metadata related to the portal, such as API products, plans, developers, applications, and more.
+Please refer to this page for more details.
```
Comment (buraksekili) [inline components/tyk-dev-portal/README.md:120]: Can you please link the page here

File: components/tyk-dev-portal/README.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+```yaml
+tyk-dev-portal:
+  extraEnvs:
+  - name: PORTAL_LOG_LEVEL
+    value: debug
+```
+
+
+## Protect Confidential Fields with Kubernetes Secrets
+In the `values.yaml` file, some fields are considered confidential, such as License , connection strings, etc.
```
Comment (buraksekili) [inline components/tyk-dev-portal/README.md:150]:
```suggestion
In the `values.yaml` file, some fields are considered confidential, such as Developer Portal license, connection strings, etc.
```

File: components/tyk-dev-portal/README.md (hunk 4 above)
Comment (komalsukhani) [reply components/tyk-dev-portal/README.md]: Ohh sorry. I forgot.
Resolved in [72084ca](https://github.com/TykTechnologies/tyk-charts/pull/197/commits/72084ca40a049655520207b89c7888f407b333e6)

**Review by buraksekili — APPROVED** [review]

**Pull request merged**